Orm/Xtensive.Orm.Tests/Issues/IssueJira0614_TypeMappingCachingInMaterializationContext.cs

```diff
@@ -257,12 +263,13 @@ public void Test01()
         }
       }
 
-      var multinodeDomainConfiguration = CreateConfiguration(typeof (Target.TimesheetCode), DomainUpgradeMode.Skip, "Model1");
+      var multinodeDomainConfiguration = CreateConfiguration(typeof(Target.TimesheetCode), DomainUpgradeMode.Skip, "Model1");
       var nodeConfiguration = CreateNodeConfiguration(Node2Name, "Model1", "Model2", DomainUpgradeMode.Skip);
+
       using (var domain = BuildDomain(multinodeDomainConfiguration)) {
-        domain.StorageNodeManager.AddNode(nodeConfiguration);
-        using (var session = domain.OpenSession()) {
-          session.SelectStorageNode(WellKnown.DefaultNodeId);
+        _ = domain.StorageNodeManager.AddNode(nodeConfiguration);
+        var selectedNode = domain.SelectStorageNode(WellKnown.DefaultNodeId);
```

Comment:
For cases like this, we would introduce `GetOrAddNode` method (see below).
```csharp
var storageNode = domain.StorageNodeManager.GetOrAddNode(nodeConfiguration);
using (var session = storageNode.OpenSession()) {
  ...
}
```

Reply:
Do you build nodes on demand and use them just after build?

Reply:
Yes, we do. On application start, we don't have any nodes added. On the first request, we build a domain and then add the corresponding node. All subsequent requests either use the existing node or add a new one on the first request to that node.

Reply:
The disadvantage of the suggested approach is that `GetOrAddNode` method takes node configuration as a parameter but the existing `GetNode` method gets just node id. I think we would exclude node id from config and pass it separately. So the sample would transform as follows.
```csharp
var storageNode = domain.StorageNodeManager.GetOrAddNode(nodeId, nodeConfiguration);
using (var session = storageNode.OpenSession()) {
  ...
}
// AddNode method should also be changed for consistency.
_ = domain.StorageNodeManager.AddNode(nodeId, nodeConfiguration);
```

Reply:
NodeConfiguration is required to have nodeId anyway, except for default node. It validates before StorageNode build

Reply:
But it is used only in the validation. The `Validate` method is called by the internal `BuildNode` method. In turn, the `BuildNode` method is called only from the `AddNode`.
As you can see we can easily transform all this chain.

Reply:
That's why I added extra check for ```nodeId``` to ```GetOrAdd```. Take a look at new new commits.

Reply:
Can we keep it this way, or even remove the `GetOrAddNode` method for now?
I'd suggest merging this PR to the 6.0 branch and to the master then.
Once the code is in the master I'd think about further improvements.

Reply:
I will keep StorageNodeManager API the same. GetOrAdd functionality can be implemented as an extension in your solution. Later we could integrate your implementation to DataObjects.Net